src/sentry/tasks/merge.py

```diff
@@ -257,11 +257,16 @@ def merge_objects(models, group, new_group, limit=1000,
                             )
                     elif model == GroupTagValue:
                         with transaction.atomic(using=router.db_for_write(model)):
```

Comment:
It's been this way since it was introduced in f50c9ebaa9f95db8c7bdfb9c5463c07b85e2bf3a, as far as I can tell.